tasks/fullStack/app/controllers/reviews_controller.rb

```diff
@@ -1,29 +1,45 @@
 class ReviewsController < ApplicationController
+  before_action :set_product, only: [:new]
 
   DEFAULT_TAGS = ['default']
 
   def index
-    if params[:shop_id].present? && Shop.where("id = #{params[:shop_id]}").present?
+    params[:shop_id] ||= 1
+    @shop = Shop.find(params[:shop_id])
+
+    if @shop.present?
       params[:per_page] ||= 10
       offset = params[:page].to_i * params[:per_page]
 
+      @shops = Shop.where.not(id: params[:shop_id]).limit(5)
+
       @data = []
-      products = Product.where("shop_id = #{params[:shop_id]}").sort_by(&:created_at)[offset..(offset + params[:per_page])]
+      products = Product.where("shop_id = #{params[:shop_id]}").limit(5).sort_by(&:created_at)[offset..(offset + params[:per_page])]
```

Comment:
This line is vulnerable to malicious external attacks, can you spot it and explain how to prevent it?

Reply:
This query uses string interpolation and therefore is at risk of SQL injection. This can be prevented by refactoring this query to be a parameterised query like so:

`Product.where(shop_id: params[:shop_id])`